test/SILGen/objc_required_designated_init.swift

```diff
@@ -1,19 +1,39 @@
 // RUN: %empty-directory(%t)
 // RUN: %target-swift-frontend -emit-module -enable-sil-ownership %S/Inputs/objc_required_designated_init_2.swift -module-name Booms -o %t/Booms.swiftmodule -import-objc-header %S/Inputs/objc_required_designated_init.h
-// RUN: %target-swift-frontend -I %t -emit-silgen -enable-sil-ownership -verify %s -import-objc-header %S/Inputs/objc_required_designated_init.h
+// RUN: %target-swift-frontend -I %t -emit-silgen -enable-sil-ownership -verify %s -import-objc-header %S/Inputs/objc_required_designated_init.h | %FileCheck %s
+// RUN: %target-swift-frontend -I %t -emit-ir %s -import-objc-header %S/Inputs/objc_required_designated_init.h
 
 // REQUIRES: objc_interop
 
 import Booms
 
-class Baboom: Boom {
-  required init() {
+class Baboom : Boom {
+  @objc dynamic required init() {
     super.init()
   }
 }
 
-class BigBadaBoom<V>: Badaboom<V> {
+class BigBadaBoom<V> : Badaboom<V> {
   required init() {
     super.init()
   }
 }
+
+class Root {
+  @objc dynamic required init() {}
```

Comment:
This was the broken case -- it's a root class so the initializer doesn't override anything, but we were still calling the allocating entry point with class_method.